### ISSUE TITLE
Callback in tf.Print to access raw data 

### ISSUE THREAD

**Issue by Overdrivr** [tensorflow/tensorflow#13964]
### System information
- **Have I written custom code**: No
- **OS**: Win 10
- **TensorFlow installed from**: binary
- **TensorFlow version**: 1.3.0
- **Python version**:  3.5
- **Bazel version (if compiling from source)**: /
- **CUDA/cuDNN version**:  CPU
- **GPU model and memory**: CPU
- **Exact command to reproduce**: /

### Feature request

The `tf.Print` op lets us print tensor values when it gets evaluated. However, it does not provide any way to access the raw data programmatically. AFAIK, there is no way to access this data during computation. Please correct me if I'm wrong

With Theano, I was able to provide a custom callback to the printing facility, to access the tensor values when tensor was evaluated. This is extremely useful for image tensors for instance to be able to plot them at evaluation time. With TF I cannot do that.

Basically, what I am proposing is adding a custom `callback` parameter to the op:
```
Print(
    input_,
    data,
    message=None,
    first_n=None,
    summarize=None,
    callback=None,
    name=None
)
```
If callback is undefined, `tf.Print` behavior remains identical. Otherwise, `tf.Print` will call the callback with `([values], message, name).

**Comment by mrry** [comment]
You can use the `tf.py_func()` op to register a Python callback with access to the raw tensor values (as a NumPy array). Simply return the arguments after executing your Python code, and it will act as an identity op.
